crates/proof-of-sql/src/base/database/owned_table_utility.rs

```diff
@@ -231,3 +235,66 @@ pub fn timestamptz<S: Scalar>(
         OwnedColumn::TimestampTZ(time_unit, timezone, data.into_iter().collect()),
     )
 }
+
+/// Creates a (Identifier, OwnedColumn) pair for a timestamp column.
+/// This is primarily intended for use in conjunction with [owned_table].
+///
+/// # Parameters
+/// - `name`: The name of the column.
+/// - `time_unit`: The time unit of the timestamps.
+/// - `timezone`: The timezone for the timestamps.
+/// - `data`: The data for the column, provided as an iterator over `i64` values representing time since the unix epoch.
+///
+/// # Example
+/// ```
+/// use proof_of_sql::base::{database::owned_table_utility::*,
+///     scalar::Curve25519Scalar,
+/// };
+/// use proof_of_sql_parser::{
+///    posql_time::{PoSQLTimeZone, PoSQLTimeUnit}};
+///
+/// let result = owned_table::<Curve25519Scalar>([
+///         timestamptz("event_time",
+///         PoSQLTimeUnit::Second,
+///         vec![
+///             "1969-12-31T23:59:59Z", // One second before the Unix epoch
+///             "1970-01-01T00:00:00Z", // The Unix epoch
+///             "1970-01-01T00:00:01Z", // One second after the Unix epoch
+///         ].iter().map(|s| s.to_string())),
+/// ]);
+/// ```
+pub fn timestamptz<S: Scalar>(
```

Comment:
Rename this one.
```suggestion
pub fn timestamptz_from_strings<S: Scalar>(
```
Also, add unit tests for this method since it is non-trivial. (And has a bug.)

Although, I think it makes more sense to remove this method altogether and just handle parsing in the test itself like this:
```rust
timestamptz_epoch("a", PoSQLTimeUnit::Millisecond, PoSQLTimeZone::Utc,
    ["1969-12-31T23:59:59Z", "1970-01-01T00:00:00Z", "1970-01-01T00:00:01Z"].iter()
        .map(|s| 
            PoSQLTimestamp::try_from(s).unwrap().timestamp.timestamp_millis()
        )

```

Reply:
This function is not creating a slice of ```PoSqlTimestamps```, it is creating an owned table. Removing this function makes it far more cumbersome to instantiate owned tables of ```timestamptz``` elsewhere in the code